rimage/man_tgl.c

```diff
@@ -0,0 +1,126 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2019 Intel Corporation. All rights reserved.
+
+#include "css.h"
+#include "cse.h"
+#include "plat_auth.h"
+#include "manifest.h"
+#include <version.h>
+
+/* manifest template */
+struct fw_image_manifest_v2_5 tgl_manifest = {
+	.cse_partition_dir_header = {
+		.header_marker = CSE_HEADER_MAKER,
+		.nb_entries = MAN_CSE_PARTS,
+		.header_version = 1,
+		.entry_version = 1,
+		.header_length = sizeof(struct CsePartitionDirHeader),
+		.partition_name = "ADSP",
+	},
+
+	.cse_partition_dir_entry = {
+		{
+			/* CssHeader + platformFirmwareAuthenticationExtension - padding */
+			.entry_name = "ADSP.man",
+			.offset = MAN_CSS_HDR_OFFSET,
+			.length = sizeof(struct css_header_v1_8) +
+				PLAT_AUTH_SIZE,
+		},
+		{	/* ADSPMetadataFileExtension */
+			.entry_name = "cavs0015.met",
+			.offset = MAN_META_EXT_OFFSET_V1_8,
+			.length = sizeof(struct sof_man_adsp_meta_file_ext_v2_5),
+		},
+		{	/* AdspFwBinaryDesc */
+			.entry_name = "cavs0015",
+			.offset = MAN_FW_DESC_OFFSET_V2_5,
+			.length = 0,	/* calculated by rimage - */
+		},
+
+	},
+
+	.css = {
+		.header_type	= MAN_CSS_MOD_TYPE,
+		.header_len	= MAN_CSS_HDR_SIZE,
+		.header_version	= MAN_CSS_HDR_VERSION,
+		.module_vendor	= MAN_CSS_MOD_VENDOR,
+		.size		= 222,
+		.header_id	= MAN_CSS_HDR_ID,
+		.padding	= 0,
+		.version = {
+			.major_version	= SOF_MAJOR,
+			.minor_version	= SOF_MINOR,
+			.hotfix_version = 0,
+			.build_version	= SOF_BUILD,
+		},
+		.modulus_size	= MAN_CSS_MOD_SIZE,
+		.exponent_size	= MAN_CSS_EXP_SIZE,
+	},
+
+	.signed_pkg = {
+		.ext_type	= SIGN_PKG_EXT_TYPE,
+		.ext_len	= sizeof(struct signed_pkg_info_ext),
+		.name		= "ADSP",
+		.vcn		= 0,
+		.bitmap		= {0, 0, 0, 0, 8},
+
+		.module[0]	= {
+			.name		= "cavs0015.met",
+			.meta_size	= 96,
+			.type		= 0x03,
+			.hash_algo	= 0x02, /* SHA 256 */
```

Comment:
SHA384 in comment ?

Reply:
This one is still sha256, sha384 is in adsp ext description